src/nfa/builder.cc

```diff
@@ -217,6 +219,48 @@ Nfa builder::create_sigma_star_nfa(mata::Alphabet* alphabet) {
     return nfa;
 }
 
+Nfa builder::create_random_nfa_tabakov_vardi(const size_t num_of_states, const size_t alphabet_size, const float transition_density, const float final_state_density) {
+    if (transition_density < 0 || static_cast<size_t>(transition_density) > num_of_states) {
```

Comment:
Isn't `transition_density` supposed to be in [0,1] interval?

Reply:
I agree that this might be confusing for the user.

Reply:
Yeah, it can be confusing. The `transition_density` can be within the range [0,num_of_states]. I'll change the name of the parameter to something more self-explanatory.